src/structs/lepton_decoder.rs

```diff
@@ -183,43 +159,30 @@ fn decode_row<R: Read>(
     qt: &QuantizationTables,
     left_model: &ProbabilityTables,
     middle_model: &ProbabilityTables,
-    right_model: &ProbabilityTables,
     image_data: &mut BlockBasedImage,
     block_context: &mut BlockContext,
     neighbor_summary_cache: &mut [NeighborSummary],
     component_size_in_blocks: i32,
     features: &EnabledFeatures,
 ) -> Result<()> {
     let block_width = image_data.get_block_width();
-    if block_width > 0 {
-        parse_token::<R, false>(
-            model,
-            bool_reader,
-            image_data,
-            block_context,
-            neighbor_summary_cache,
-            qt,
-            left_model,
-            features,
-        )
-        .context(here!())?;
-        let offset = block_context.next();
 
-        if offset >= component_size_in_blocks {
```

Comment:
You need to reintroduce this - it is the case of truncated file, blocks can be missing from any point inside the line, as far as I understand.